### PR TITLE
Add `isNotEmpty` to `Table` assertions

### PR DESCRIPTION
**Pull request by mk868**
The `TableAssert` currently contains `isEmpty()` assertion, but no `isNotEmpty()`. I want to extend this to be like other collection assertions.

#### Check List:
* Unit tests : YES
* Javadoc with a code example (on API only) : YES
* PR meets the [contributing guidelines](https://github.com/assertj/assertj/blob/main/CONTRIBUTING.md)

#### Changes in this PR:
* A new assertion `isNotEmpty()` added to the `TableAssert` class
* Added unit tests for `TableAssert#isNotEmpty()`
* Fixed typo in the `TableAssert_isEmpty_Test#should_fail_if_actual_is_null()` test method

### CODE REVIEW

**Comment by mk868** [comment]
Branch rebased

**Comment by mk868** [comment]
Branch rebased

**Comment by mk868** [comment]
Is there any chance to merge this?
Or should I create an issue with a better described motivation of this change?

**Comment by scordio** [comment]
Hi @mk868, sorry for the slow feedback! I just let the pipeline run, you might face some formatting alerts that can be fixed with spotless.

Anyway, I'll jump on the PR over the weekend and get back to you.

File: assertj-tests/assertj-integration-tests/assertj-guava-tests/src/test/java/org/assertj/tests/guava/api/TableAssert_isNotEmpty_Test.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+  @Test
+  void should_fail_if_actual_is_null() {
+    // GIVEN
+    actual = null;
+    // WHEN
+    Throwable throwable = catchThrowable(() -> assertThat(actual).isNotEmpty());
+    // THEN
+    assertThat(throwable).isInstanceOf(AssertionError.class)
+      .hasMessage(actualIsNull());
```
Comment (scordio) [inline assertj-tests/assertj-integration-tests/assertj-guava-tests/src/test/java/org/assertj/tests/guava/api/TableAssert_isNotEmpty_Test.java:42]: We favor BDD style assertions for our own tests, i.e., :

```suggestion
    then(throwable).isInstanceOf(AssertionError.class)
      .hasMessage(actualIsNull());
```

From `org.assertj.core.api.BDDAssertions.then`

File: assertj-tests/assertj-integration-tests/assertj-guava-tests/src/test/java/org/assertj/tests/guava/api/TableAssert_isNotEmpty_Test.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  }
+
+  @Test
+  void should_return_this() {
+    // GIVEN
+    TableAssert<Integer, Integer, String> assertion = assertThat(actual);
+    // WHEN
+    TableAssert<Integer, Integer, String> returnedAssertion = assertion.isNotEmpty();
+    // THEN
+    assertThat(returnedAssertion).isSameAs(assertion);
```
Comment (scordio) [inline assertj-tests/assertj-integration-tests/assertj-guava-tests/src/test/java/org/assertj/tests/guava/api/TableAssert_isNotEmpty_Test.java:63]:
```suggestion
    then(returnedAssertion).isSameAs(assertion);
```

File: assertj-tests/assertj-integration-tests/assertj-guava-tests/src/test/java/org/assertj/tests/guava/api/TableAssert_isNotEmpty_Test.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import org.junit.jupiter.api.Test;
+
+/**
+ * @author Maciej Kucharczyk
+ */
+class TableAssert_isNotEmpty_Test extends TableAssertBaseTest {
+
+  @Test
+  void should_pass_if_actual_is_not_empty() {
+    assertThat(actual).isNotEmpty();
```
Comment (scordio) [inline assertj-tests/assertj-integration-tests/assertj-guava-tests/src/test/java/org/assertj/tests/guava/api/TableAssert_isNotEmpty_Test.java:31]:
```suggestion
    // WHEN/THEN
    assertThat(actual).isNotEmpty();
```

**Review by scordio — COMMENTED** [review]
Just a few cosmetic comments. No worries if you don't have time, I can take care of them during the merge.

**Comment by scordio** [comment]
Thanks for your first contribution, @mk868!

**Pull request merged**